### ISSUE TITLE
Firefox support?

### ISSUE THREAD

**Issue by wldcordeiro** [ducky/play-midnight#48]
Now that Firefox uses webextensions this really should be easier to maintain for both browsers.

**Comment by chrisxclash** [comment]
Yeah it's definitely on my list. It's mostly 80%-ish the way there, but there were some issues that I still need to look into that I mentioned in a comment on this PR: #42

**Comment by phasze** [comment]
Any update on this?

**Comment by ducky** [comment]
@phasze - Sadly no. I’ve been busy with a few other projects so I’ve mainly been in maintenance mode with Play Midnight the last year or two. I’m always open to other people PRing it if they wanna tinker.

**Comment by phasze** [comment]
Alright I'll take a look today. I swapped to firefox recently because of some of chrome's policy changes and have been missing your lovely extension. 

I built your project and loaded it as a firefox addon.  Working great so far.  I see the PR you linked had a few issues, are those still the outstanding ones? Anything else I should look out for?

**Comment by ducky** [comment]
Thanks for looking into it. As far as I know those are the main issues. I'll be honest I haven't looked at Play Music (especially Firefox) in ages since I use Spotify as my main driver now.

I know the biggest issue was the first one I mentioned which was saving. Rather than adopt what Chrome has working with callbacks, Firefox opted to use promises which made working them together a pain.

I can't say if those three issues are the only ones anymore, so I'm all ears, haha.

**Comment by phasze** [comment]
I don't see the homepage scrolling issue.

The favicon swapping does appear to still be an issue.  From a quick glance at the <link> attribute in the inspector I suspect this is because the file doesn't exist in the location it tries to set.

As for saving, I assume this is talking about saving the user preferences when they click on the gear icon and change something?  From my quick test that appears to be working as well.

**Comment by ducky** [comment]
- Scrolling: Cool, I don't even remember what the actual issue is anymore.
- Favicon: Yeah, I think it was just a disconnect to where the background pages put things. Most likely the way the url is referenced. I never looked much into it.
- Saving: I believe it's the sync settings. You'd have to check it between two firefox browsers (two different machines) to make sure the settings are synced across all browsers. I think I use local storage as a backup, maybe?
